modified_streusle/conllulex2json.py

```diff
@@ -0,0 +1,416 @@
+                if lexe['ss2'] is not None:
+                    lexe['ss2'] = ss_mapper(lexe['ss2'])
+                assert all(t>0 for t in lexe['toknums']),('Token offsets must be positive',lexe)
+            if 'wmwes' in sent:
+                for lexe in sent['wmwes'].values():
+                    assert all(t>0 for t in lexe['toknums']),('Token offsets must be positive',lexe)
+
+            if not morph_syn:
+                for tok in sent['toks']:
+                    tok.pop('feats', None)
+                    tok.pop('head', None)
+                    tok.pop('deprel', None)
+                    tok.pop('edeps', None)
+
+            if not misc:
+                for tok in sent['toks']:
+                    tok.pop('misc', None)
+
+            yield sent
+        return
+
+    # Otherwise, .conllulex: create data structures and check consistency
+
+    lc_tbd = 0
+
+    def _postproc_sent(sent):
+        nonlocal lc_tbd
+
+        # check that tokens are numbered from 1, in order
+        for i,tok in enumerate(sent['toks'], 1):
+            assert tok['#']==i
+
+        # check that MWEs are numbered from 1 based on first token offset
+        xmwes =  [(e["toknums"][0], 's', mwenum) for mwenum,e in sent['smwes'].items()]
+        xmwes += [(e["toknums"][0], 'w', mwenum) for mwenum,e in sent['wmwes'].items()]
+        xmwes.sort()
+        for k,mwe in chain(sent['smwes'].items(), sent['wmwes'].items()):
+            assert int(k)-1<len(xmwes),f"In {sent['sent_id']}, MWE index {k} exceeds number of MWEs in the sentence"
+            assert xmwes[int(k)-1][2]==k,f"In {sent['sent_id']}, MWEs are not numbered in the correct order: use normalize_mwe_numbering.py to fix"
+
+        # check that lexical & weak MWE lemmas are correct
+        lexes_to_validate = chain(sent['swes'].values(), sent['smwes'].values()) if validate_type else []
+        for lexe in lexes_to_validate:
+            assert lexe['lexlemma']==' '.join(sent['toks'][i-1]['lemma'] for i in lexe['toknums']),f"In {sent['sent_id']}, MWE lemma is incorrect: {lexe} vs. {sent['toks'][lexe['toknums'][0]-1]}"
+            lc = lexe['lexcat']
+            if lc.endswith('!@'): lc_tbd += 1
+            valid_ss = supersenses_for_lexcat(lc)
+            # change for PASTRIE--don't check lexcat
+            #if lc=='V':
+            #    assert len(lexe['toknums'])==1,f'In {sent["sent_id"]}, Verbal MWE "{lexe["lexlemma"]}" lexcat must be subtyped (V.VID, etc., not V)'
+            ss, ss2 = lexe['ss'], lexe['ss2']
+            if valid_ss:
+                if ss=='??':
+                    assert ss2 is None
+                elif ss is not None and (ss not in valid_ss or (lc in ('N','V') or lc.startswith('V.'))!=(ss2 is None) or (ss2 is not None and ss2 not in valid_ss)):
+                    assert False,f"In {sent['sent_id']}, invalid supersense(s) in lexical entry: {lexe}"
+                elif ss is not None and ss.startswith('p.'):
```

Comment:
I think `ss is not None` should be removed from the above two `elif` conditions. This is allowing tokens missing a supersense to slip through.